From 686c0111e151ca2045a3829608962320a9bc1562 Mon Sep 17 00:00:00 2001
From: Fabrice Gasnier <fabrice.gasnier@foss.st.com>
Date: Mon, 17 Mar 2025 11:27:40 +0100
Subject: [PATCH] arm64: dts: st: lptimer: adopt fallback compatibles on
 stm32mp251

Align with upstream recommendation to adopt fallback/backward compatible
strings on stm32mp25 [1].

[1] https://lore.kernel.org/linux-arm-kernel/20250224180150.3689638-1-fabrice.gasnier@foss.st.com/

Signed-off-by: Fabrice Gasnier <fabrice.gasnier@foss.st.com>
Change-Id: I1852976d4654d790e65a8157883755623723b9e3
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/445389
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Olivier MOYSAN <olivier.moysan@foss.st.com>
---
 arch/arm64/boot/dts/st/stm32mp251.dtsi | 50 +++++++++++++-------------
 1 file changed, 25 insertions(+), 25 deletions(-)

--- a/arch/arm64/boot/dts/st/stm32mp251.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp251.dtsi
@@ -737,7 +737,7 @@
 			};
 
 			lptimer1: timer@40090000 {
-				compatible = "st,stm32mp25-lptimer";
+				compatible = "st,stm32mp25-lptimer", "st,stm32-lptimer";
 				reg = <0x40090000 0x400>;
 				interrupts-extended = <&exti1 47 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_LPTIM1>;
@@ -750,30 +750,30 @@
 				status = "disabled";
 
 				counter {
-					compatible = "st,stm32mp25-lptimer-counter";
+					compatible = "st,stm32mp25-lptimer-counter", "st,stm32-lptimer-counter";
 					status = "disabled";
 				};
 
 				pwm {
-					compatible = "st,stm32mp25-pwm-lp";
+					compatible = "st,stm32mp25-pwm-lp", "st,stm32-pwm-lp";
 					#pwm-cells = <3>;
 					status = "disabled";
 				};
 
 				timer {
-					compatible = "st,stm32mp25-lptimer-timer";
+					compatible = "st,stm32mp25-lptimer-timer", "st,stm32-lptimer-timer";
 					status = "disabled";
 				};
 
 				trigger@0 {
-					compatible = "st,stm32mp25-lptimer-trigger";
+					compatible = "st,stm32mp25-lptimer-trigger", "st,stm32-lptimer-trigger";
 					reg = <0>;
 					status = "disabled";
 				};
 			};
 
 			lptimer2: timer@400a0000 {
-				compatible = "st,stm32mp25-lptimer";
+				compatible = "st,stm32mp25-lptimer", "st,stm32-lptimer";
 				reg = <0x400a0000 0x400>;
 				interrupts-extended = <&exti1 48 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_LPTIM2>;
@@ -786,23 +786,23 @@
 				status = "disabled";
 
 				counter {
-					compatible = "st,stm32mp25-lptimer-counter";
+					compatible = "st,stm32mp25-lptimer-counter", "st,stm32-lptimer-counter";
 					status = "disabled";
 				};
 
 				pwm {
-					compatible = "st,stm32mp25-pwm-lp";
+					compatible = "st,stm32mp25-pwm-lp", "st,stm32-pwm-lp";
 					#pwm-cells = <3>;
 					status = "disabled";
 				};
 
 				timer {
-					compatible = "st,stm32mp25-lptimer-timer";
+					compatible = "st,stm32mp25-lptimer-timer", "st,stm32-lptimer-timer";
 					status = "disabled";
 				};
 
 				trigger@1 {
-					compatible = "st,stm32mp25-lptimer-trigger";
+					compatible = "st,stm32mp25-lptimer-trigger", "st,stm32-lptimer-trigger";
 					reg = <1>;
 					status = "disabled";
 				};
@@ -2049,7 +2049,7 @@
 			};
 
 			lptimer3: timer@46050000 {
-				compatible = "st,stm32mp25-lptimer";
+				compatible = "st,stm32mp25-lptimer", "st,stm32-lptimer";
 				reg = <0x46050000 0x400>;
 				interrupts-extended = <&exti2 29 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_LPTIM3>;
@@ -2061,30 +2061,30 @@
 				status = "disabled";
 
 				counter {
-					compatible = "st,stm32mp25-lptimer-counter";
+					compatible = "st,stm32mp25-lptimer-counter", "st,stm32-lptimer-counter";
 					status = "disabled";
 				};
 
 				pwm {
-					compatible = "st,stm32mp25-pwm-lp";
+					compatible = "st,stm32mp25-pwm-lp", "st,stm32-pwm-lp";
 					#pwm-cells = <3>;
 					status = "disabled";
 				};
 
 				timer {
-					compatible = "st,stm32mp25-lptimer-timer";
+					compatible = "st,stm32mp25-lptimer-timer", "st,stm32-lptimer-timer";
 					status = "disabled";
 				};
 
 				trigger@2 {
-					compatible = "st,stm32mp25-lptimer-trigger";
+					compatible = "st,stm32mp25-lptimer-trigger", "st,stm32-lptimer-trigger";
 					reg = <2>;
 					status = "disabled";
 				};
 			};
 
 			lptimer4: timer@46060000 {
-				compatible = "st,stm32mp25-lptimer";
+				compatible = "st,stm32mp25-lptimer", "st,stm32-lptimer";
 				reg = <0x46060000 0x400>;
 				interrupts-extended = <&exti2 30 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_LPTIM4>;
@@ -2096,30 +2096,30 @@
 				status = "disabled";
 
 				counter {
-					compatible = "st,stm32mp25-lptimer-counter";
+					compatible = "st,stm32mp25-lptimer-counter", "st,stm32-lptimer-counter";
 					status = "disabled";
 				};
 
 				pwm {
-					compatible = "st,stm32mp25-pwm-lp";
+					compatible = "st,stm32mp25-pwm-lp", "st,stm32-pwm-lp";
 					#pwm-cells = <3>;
 					status = "disabled";
 				};
 
 				timer {
-					compatible = "st,stm32mp25-lptimer-timer";
+					compatible = "st,stm32mp25-lptimer-timer", "st,stm32-lptimer-timer";
 					status = "disabled";
 				};
 
 				trigger@3 {
-					compatible = "st,stm32mp25-lptimer-trigger";
+					compatible = "st,stm32mp25-lptimer-trigger", "st,stm32-lptimer-trigger";
 					reg = <3>;
 					status = "disabled";
 				};
 			};
 
 			lptimer5: timer@46070000 {
-				compatible = "st,stm32mp25-lptimer";
+				compatible = "st,stm32mp25-lptimer", "st,stm32-lptimer";
 				reg = <0x46070000 0x400>;
 				interrupts-extended = <&exti2 31 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_LPTIM5>;
@@ -2131,23 +2131,23 @@
 				status = "disabled";
 
 				counter {
-					compatible = "st,stm32mp25-lptimer-counter";
+					compatible = "st,stm32mp25-lptimer-counter", "st,stm32-lptimer-counter";
 					status = "disabled";
 				};
 
 				pwm {
-					compatible = "st,stm32mp25-pwm-lp";
+					compatible = "st,stm32mp25-pwm-lp", "st,stm32-pwm-lp";
 					#pwm-cells = <3>;
 					status = "disabled";
 				};
 
 				timer {
-					compatible = "st,stm32mp25-lptimer-timer";
+					compatible = "st,stm32mp25-lptimer-timer", "st,stm32-lptimer-timer";
 					status = "disabled";
 				};
 
 				trigger@4 {
-					compatible = "st,stm32mp25-lptimer-trigger";
+					compatible = "st,stm32mp25-lptimer-trigger", "st,stm32-lptimer-trigger";
 					reg = <4>;
 					status = "disabled";
 				};
